packages/edit-site/src/components/page-templates/template-actions.js

```diff
@@ -89,29 +106,82 @@ export const deleteTemplateAction = {
 	isPrimary: true,
 	icon: trash,
 	isEligible: isTemplateRemovable,
+	supportsBulk: true,
 	hideModalHeader: true,
-	RenderModal: ( { item: template, closeModal } ) => {
+	RenderModal: ( { items: templates, closeModal, onPerform } ) => {
 		const { removeTemplate } = useDispatch( editSiteStore );
+		const { createSuccessNotice, createErrorNotice } =
+			useDispatch( noticesStore );
+		const { deleteEntityRecord } = useDispatch( coreStore );
 		return (
 			<VStack spacing="5">
 				<Text>
-					{ sprintf(
-						// translators: %s: The template or template part's title.
-						__( 'Are you sure you want to delete "%s"?' ),
-						decodeEntities( template.title.rendered )
-					) }
+					{ templates.length > 1
+						? sprintf(
+								// translators: %s: The template or template part's title.
+								__(
+									'Are you sure you want to delete %s items?'
+								),
+								templates.length
+						  )
+						: sprintf(
+								// translators: %s: The template or template part's title.
+								__( 'Are you sure you want to delete "%s"?' ),
+								decodeEntities(
+									templates && templates[ 0 ]?.title?.rendered
+								)
+						  ) }
 				</Text>
 				<HStack justify="right">
 					<Button variant="tertiary" onClick={ closeModal }>
 						{ __( 'Cancel' ) }
 					</Button>
 					<Button
 						variant="primary"
-						onClick={ () =>
-							removeTemplate( template, {
-								allowUndo: false,
-							} )
-						}
+						onClick={ async () => {
+							if ( templates.length > 1 ) {
+								try {
+									await Promise.allSettled(
+										templates.map( ( template ) => {
+											return deleteEntityRecord(
+												'postType',
+												template.type,
+												template.id,
+												{ force: true },
+												{ throwOnError: true }
+											);
+										} )
+									);
+									createSuccessNotice(
+										__(
+											'The selected items were deleted with success.'
+										),
+										{
+											type: 'snackbar',
+											id: 'edit-site-page-trashed',
+										}
+									);
+								} catch ( error ) {
+									const errorMessage =
+										error.message &&
+										error.code !== 'unknown_error'
+											? error.message
+											: __(
+													'An error occurred while deleting the items.'
+											  );
+
+									createErrorNotice( errorMessage, {
+										type: 'snackbar',
+									} );
+								}
+							} else {
+								await removeTemplate( templates[ 0 ], {
+									allowUndo: false,
+								} );
+							}
```

Comment:
Hi @mcsf,
Because the removeTemplate takes care of error handling displaying the success notices etc... We only need to call it and everything is handled by the action.
We would have two ways to remove error handling from here:
-  Call removeTemplate in a loop. - The use experience would not be great instead a single notice saying x items were removed we would have a notice per item.
- Implement a removeTemplates action or expand removeTemplate to support multiple templates - Also not a great option we don't have any other case yet where we want to remove multiple templates so I'm not sure exposing an action for that is a good idea for now,

Given that the two approaches were not great I opted for doing the error handling inline.

Reply:
Yeah, mine was a leading question, it was more about: can we take the time to create some symmetry to keep the codebase less surprising? For example:

* Create a new action, `removeTemplates`, but keep it private using the `private-apis` module.
* Refactor the preexisting `removeTemplate` to simply call the new one. We see this pattern across Gutenberg (`replaceBlock` -> `replaceBlocks`).
* The new `removeTemplates` should consider `templates.length` in the strings it generates for success and error notices.

I'm happy to be challenged on this suggestion, btw.

Reply:
I followed the suggestion and introduced a removeTemplates private API 👍